jtravis/src/main/java/fr/inria/spirals/jtravis/helpers/BuildHelper.java

```diff
@@ -185,29 +185,88 @@ private static void getBuildsFromSlugRecursively(String slug, List<Build> result
                 }
             }
 
-
-
             if (buildArray.size() == 0) {
                 dateReached = true;
             }
 
             if (limitDate != null && !dateReached) {
-                getBuildsFromSlugRecursively(slug, result, limitDate, lastBuildNumber, eventTypes, limitNumber, status, prNumber, onlyAfterNumber, previousBranch);
+                getBuildsFromSlugRecursively(slug, result, limitDate, lastBuildNumber, original_after_number, eventTypes, limitNumber, status, prNumber, onlyAfterNumber, previousBranch);
             } else {
                 eventTypes.remove(0);
                 if (!onlyAfterNumber) {
                     after_number = 0;
+                } else {
+                    after_number = original_after_number;
+                }
+                getBuildsFromSlugRecursively(slug, result, limitDate, after_number, original_after_number, eventTypes, limitNumber, status, prNumber, onlyAfterNumber, previousBranch);
+            }
+        } catch (IOException e) {
+            getInstance().getLogger().warn("Error when getting list of builds from slug "+slug+" : "+e.getMessage());
+        }
+    }
+
+    public static int getTheLastBuildNumberOfADate(String slug, Date date, int after_number, boolean onlyAfterNumber) {
```

Comment:
You should add some javadoc here too. And as several parts of this method came from the previous one, it could be great to factorize some elements: for example the resourceUrl, the check for after_number, ...